src/test/java/com/virtudoc/web/AppointmentRepositoryTests.java

```diff
@@ -0,0 +1,45 @@
+package com.virtudoc.web;
+
+import com.virtudoc.web.entity.Appointment;
+import com.virtudoc.web.repository.AppointmentRepository;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.annotation.Rollback;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@Rollback(false)
+public class AppointmentRepositoryTests {
+
+    @Autowired
+    private AppointmentRepository repo;
+
+    @Test
+    public void testAddNew() {
+        Appointment apt = new Appointment();
+        apt.setEmail("test@test.com");
+        apt.setPatientName("Patient 1");
+        apt.setSymptoms("Covid");
+        apt.setDoctorName("Doctor 1");
+        apt.setLocation("Kennesaw");
+        apt.setDate("Feb 3");
+        apt.setReasonForVisit("Reason for Visit");
+        Appointment savedApt = repo.save(apt);
+        Assertions.assertThat(savedApt).isNotNull();
+        Assertions.assertThat(savedApt.getAppointmentId()).isGreaterThan(0);
+    }
+
+    @Test
+    public void testListAll() {
+        Iterable<Appointment> apts = repo.findAll();
+        Assertions.assertThat(apts).hasSizeGreaterThan(0);
```

Comment:
I think this is the test with the test failure. Typically tests should create their own data needed for the test, and not rely on other tests. To speed up testing on Github Actions, all tests run asynchronously on multiple threads, so tests can't rely on the existing database being in a certain state, or being able to run before/after certain tests. `repo.save(new Appointment());` on L37 would probably be enough here.